## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#8d09fae6-3f46-4951-a57b-727f57bc03fe) |
| mint-zoned.sh | :x: [more...](#aad6df29-280d-4e25-9f81-538661d980bf) |
<a name="8d09fae6-3f46-4951-a57b-727f57bc03fe"></a>
### 10508-d00f8ba/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:30255
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 80e178b5ad8d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 1 minutes and 4 seconds
(4/15) Running aws-sdk-ruby tests ... done in 20 seconds
(5/15) Running awscli tests ... done in 2 minutes and 8 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 53 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 40 seconds
(9/15) Running minio-go tests ... done in 7 minutes and 2 seconds
(10/15) Running minio-java tests ... done in 10 minutes and 21 seconds
(11/15) Running minio-js tests ... done in 2 minutes and 48 seconds
(12/15) Running minio-py tests ... FAILED in 17 minutes and 58 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-2100a930-0716-4be6-99d4-7456b3aea6f2",
    "object_name": "0bbbb929-c818-4d66-b311-9bf5b0e01150",
    "presigned_get_object_url": "http://minio-dev6.minio.io:30255/minio-py-test-2100a930-0716-4be6-99d4-7456b3aea6f2/0bbbb929-c818-4d66-b311-9bf5b0e01150?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioazure%2F20200917%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200917T185154Z&X-Amz-Expires=120&X-Amz-SignedHeaders=host&X-Amz-Signature=49d1fec85b10b3f1593a53d15742fccf1a2122765fa3402b6ee2b9a24e5c0247",
    "response.status": 500,
    "response.reason": "Internal Server Error",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"473\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"etag\": [\"ETag\", \"\\\"f694e2b345ce50ed6cc481175f65d84c\\\"\"], \"last-modified\": [\"Last-Modified\", \"Thu, 17 Sep 2020 18:51:54 GMT\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2020-09-17T17-54-41Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"1635A6DE07CDAD5F\"], \"x-cache\": [\"X-Cache\", \"HIT\"], \"x-cache-lookup\": [\"X-Cache-Lookup\", \"HIT\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Thu, 17 Sep 2020 18:51:55 GMT\"]}}",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InternalError</Code><Message>We encountered an internal error, please try again.</Message><Key>0bbbb929-c818-4d66-b311-9bf5b0e01150</Key><BucketName>minio-py-test-2100a930-0716-4be6-99d4-7456b3aea6f2</BucketName><Resource>/minio-py-test-2100a930-0716-4be6-99d4-7456b3aea6f2/0bbbb929-c818-4d66-b311-9bf5b0e01150</Resource><RequestId>1635A6DE07CDAD5F</RequestId><HostId>5792e0f4-0129-49f8-ab2d-0b2aa6635ebf</HostId></Error>"
  },
  "message": "InternalError: message: We encountered an internal error. Please try again.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1349, in test_presigned_get_object_expiry\n    object_name).get_exception()\nminio.error.InternalError: InternalError: message: We encountered an internal error. Please try again.\n",
  "duration": 1294
}
(12/15) Running s3cmd tests ... done in 2 minutes and 16 seconds
(13/15) Running s3select tests ... FAILED in 12 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "bucket_name": "s3select-test-f40d7453-ffac-4e63-a118-5caeb91f6325"
  },
  "duration": 11777,
  "message": "Test test_13 unexpectedly failed with: 'event-type'",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/s3select/csv.py\", line 39, in test_sql_api\n    for d in data.stream(10*1024):\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select/reader.py\", line 211, in stream\n    message = self.__extract_message()\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select/reader.py\", line 161, in __extract_message\n    event_type = header_map[\"event-type\"]\nKeyError: 'event-type'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 52, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 119, in test_csv_input_custom_quote_char\n    data, sql_opts, expected_output)\n  File \"/mint/run/core/s3select/csv.py\", line 45, in test_sql_api\n    'Test {} unexpectedly failed with: {}'.format(test_name, select_err))\nValueError: Test test_13 unexpectedly failed with: 'event-type'\n",
  "status": "FAIL"
}
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
<a name="aad6df29-280d-4e25-9f81-538661d980bf"></a>
### 10508-d00f8ba/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30714
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 3013c029e7d1:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 3 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 45 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 17 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 36 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 5 seconds
(9/15) Running minio-go tests ... done in 2 minutes and 33 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 53 seconds
(11/15) Running minio-js tests ... done in 50 seconds
(12/15) Running minio-py tests ... done in 3 minutes and 31 seconds
(13/15) Running s3cmd tests ... FAILED in 21 seconds
{
  "name": "s3cmd",
  "duration": "10888",
  "function": "test_sync_list_objects",
  "status": "FAIL",
  "error": "WARNING: Bucket is not empty. Removing all the objects from it first. This may take some time...\nERROR: S3 error: 404 (Not Found)"
}
(13/15) Running s3select tests ... done in 9 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10508-d00f8ba

--- a/cmd/dynamic-timeouts.go
+++ b/cmd/dynamic-timeouts.go
@@ -28,6 +28,7 @@ const (
 	dynamicTimeoutDecreaseThresholdPct = 0.10 // Lower threshold for failures in order to decrease timeout
 	dynamicTimeoutLogSize              = 16
 	maxDuration                        = math.MaxInt64
+	maxDynamicTimeout                  = 24 * time.Hour // Never set timeout bigger than this.
 )
 
 // timeouts that are dynamically adapted based on actual usage results
@@ -113,6 +114,10 @@ func (dt *dynamicTimeout) adjust(entries [dynamicTimeoutLogSize]time.Duration) {
 		// We are hitting the timeout too often, so increase the timeout by 25%
 		timeout := atomic.LoadInt64(&dt.timeout) * 125 / 100
 
+		// Set upper cap.
+		if timeout > int64(maxDynamicTimeout) {
+			timeout = int64(maxDynamicTimeout)
+		}
 		// Safety, shouldn't happen
 		if timeout < dt.minimum {
 			timeout = dt.minimum

--- a/cmd/dynamic-timeouts_test.go
+++ b/cmd/dynamic-timeouts_test.go
@@ -194,7 +194,7 @@ func TestDynamicTimeoutAdjustExponential(t *testing.T) {
 
 	timeout := newDynamicTimeout(time.Minute, time.Second)
 
-	rand.Seed(time.Now().UTC().UnixNano())
+	rand.Seed(0)
 
 	initial := timeout.Timeout()
 
@@ -214,7 +214,7 @@ func TestDynamicTimeoutAdjustNormalized(t *testing.T) {
 
 	timeout := newDynamicTimeout(time.Minute, time.Second)
 
-	rand.Seed(time.Now().UTC().UnixNano())
+	rand.Seed(0)
 
 	initial := timeout.Timeout()
 

--- a/cmd/global-heal.go
+++ b/cmd/global-heal.go
@@ -31,7 +31,7 @@ const (
 	leaderLockTimeoutSleepInterval = time.Hour
 )
 
-var leaderLockTimeout = newDynamicTimeout(30*time.Second, 5*time.Second)
+var leaderLockTimeout = newDynamicTimeout(1*time.Minute, 30*time.Second)
 
 // NewBgHealSequence creates a background healing sequence
 // operation which crawls all objects and heal them.

--- a/cmd/server-main.go
+++ b/cmd/server-main.go
@@ -231,7 +231,7 @@ func initSafeMode(ctx context.Context, newObject ObjectLayer) (err error) {
 	}
 
 	// allocate dynamic timeout once before the loop
-	configLockTimeout := newDynamicTimeout(3*time.Second, 1*time.Second)
+	configLockTimeout := newDynamicTimeout(5*time.Second, 3*time.Second)
 
 	// ****  WARNING ****
 	// Migrating to encrypted backend should happen before initialization of any

